### DISCUSSION TITLE
How can we specify replica count when defining a Persistent Volume Claim?

### DISCUSSION THREAD

**Discussion by iosifnicolae2** [longhorn/longhorn#3869] [Q&A]
Hi.

How can we specify replica count when defining a Persistent Volume Claim?

Thanks!

**Comment by iosifnicolae2** [comment]
The single way to programatically define replica count is by defining a Storage Class?

Storage Classes are immutable and we would like to be able to set the replica count from our kubernetes manifests.

It would be awesome if we can override `longhorn.io/v1beta1/Volume` - [NumberOfReplicas](https://github.com/longhorn/longhorn-manager/blob/14c186f8b9545d62d0997d1cbd7bfc09379f3cfe/k8s/pkg/apis/longhorn/v1beta2/volume.go#L195) by passing the value as annotation on PVC.

**Reply by joshimoo** [reply]
Currently we don't support volume creation parameters via annotations, can you open a feature request for this use case :)
As a workaround is it sufficient to provide multiple storage classes (1 class for each valid parameter set) then do the storage class selection in your manifests?

As part of the feature request can you outline your use case for the parameters and how much dynamism you require?
Do you have an operator or something that creates these pvc manifests dynamically and therefore requires control of this paramter directly?

**Reply by iosifnicolae2** [reply]
We're deploying our applications using helm charts and it would be awesome to be able to define the number of replicas for each volume directly from the helm chart (also, some customers might want to have greater redundancy or some services can be converted from non-HA to HA versions, so we need to reduce the number of volume replicas from 3 to 1).
